### DISCUSSION TITLE
HasCollection api source code question

### DISCUSSION THREAD

**Discussion by ArenaSu** [milvus-io/milvus#33367] [Q&A and General discussion] [answered]
When i read HasCollection api source code(master branch), the flow is handleHasCollection[internal/distributed/proxy/httpserver/handler.go]->HasCollection[milvuspb/milvus.pb.go]->HasCollection[internal/proxy/impl.go]->node.sched.ddQueue.Enqueue(hct)[internal/proxy/impl.go].And taskScheduler schedules the gRPC tasks, but milvus architecture[https://milvus.io/static/a85f4600a1897497e9a946ea806d20d9/c0f46/milvus_architecture.png] shows proxy directly send request to query node.I'm confused about it.And the hasCollectionTask exec t.rootCoord.HasCollection.It means proxy send request to root coord.

**Comment by ArenaSu** [comment]
Or maybe anyone can tell me DDL/DCL、DML、search/query each include which operations.

**Comment by yhmo** [comment, marked as answer]
node.sched.ddQueue.Enqueue(hct)
The "hct" is a hasCollectionTask. In hasCollectionTask.Execute(), it calls rootCoord.HasCollection()

DDL ---- createCollection/createPartition/dropCollection/dropPartiton/hasCollection/hasPartition
DML ---- insert/delete/upsert
DQL ---- search/query

**Reply by xiaofan-luan** [reply]
DDL - data definition 
DML - data mutation
DQL - data query
